src/fudge/log.clj

```diff
@@ -52,12 +61,15 @@
       {:message data}))
 
 (defn prepare-data-for-logging
-  "Prepare the `data` structure being logged by setting the `:level`
-   and `:date` values"
-  [config data level]
+  "Prepare the `data` structure being logged by setting the `:level`,
+   `:date`, `:ns`, `:file` and `:line` values"
+  [config data level calling-meta]
   (->> data
        normalize-data
        (merge {:date (call config :date-fn)
+               :ns (:ns calling-meta)
+               :file (:file calling-meta)
+               :line (:line calling-meta)
```

Comment:
this could be `(merge {:date... :level...} calling-meta)` ?

I'm wondering if the calling-meta should be a) bundled in a sub-key, and/or b) passed within `data`.